packages/ember-metal/lib/property_events.js

```diff
@@ -252,14 +251,49 @@ function changeProperties(callback, binding) {
   }
 }
 
+function indexOf(array, target, method) {
+  let index = -1;
+  // hashes are added to the end of the event array
+  // so it makes sense to start searching at the end
+  // of the array and search in reverse
+  for (let i = array.length - 3; i >= 0; i -= 3) {
+    if (target === array[i] && method === array[i + 1]) {
+      index = i;
+      break;
+    }
+  }
+  return index;
+}
+
+function accumulateListeners(obj, eventName, otherActions, meta) {
+  if (meta === null || meta === undefined) { return; }
```

Comment:
you can even remove meta existence check here since its always passed and its private api